Property 'speedAccuracy' not found on object of type 'CLLocation *'

## 🐛 Bug Report

# LocationManager.m
`    if (@available(iOS 10.0, *)) {
        speedAccuracy = location.speedAccuracy;
    }
`
<!--- Summary description of the bug --->

### Expected behavior

### Reproduction steps

### Configuration

**Version:** 6.1.4

**Platform:** 
- [ ] :iphone: iOS
- [ ] :robot: Android

I have the same bug after updating lib:

`geolocator: ^6.1.4
  geocoding: ^1.0.5`

and the problem is:

`Running Xcode build...
Xcode build done.                                           24,5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Library/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-6.1.4/ios/Classes/Utils/LocationMapper.m:20:34: error: property 'speedAccuracy' not found on object of type 'CLLocation *'
            speedAccuracy = location.speedAccuracy;
                                     ^
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'in_app_purchase' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'in_app_purchase' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro Max.`

Thanks

I installed the [latest package](https://pub.dev/packages/geolocator) i.e geolocator: ^6.1.4. 
I was able to use it to get my current position on the android device but once I ran the iOS simulator, I got the following error:

_Launching lib/main.dart on iPhone 11 Pro Max in debug mode...
lib/main.dart:1
Xcode build done.                                           12.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    /Users/mac/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-6.1.4/ios/Classes/Utils/LocationMapper.m:20:34: error: property 'speedAccuracy' not found on object of type 'CLLocation *'
            speedAccuracy = location.speedAccuracy;
                                     ^
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description
Could not build the application for the simulator.
Error launching application on iPhone 11 Pro Max.
Exited (sigterm)_

This is my environment; 
_Flutter 1.22.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8874f21e79 (6 days ago) • 2020-10-29 14:14:35 -0700
Engine • revision a1440ca392
Tools • Dart 2.10.3_

--- a/geolocator/CHANGELOG.md
+++ b/geolocator/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 6.1.4+1
+
+- Hotfix to make sure the `CLLocation.speedAccuracy` property is only compiled when using Xcode 12 or higher (see [#577](https://github.com/Baseflow/flutter-geolocator/issues/577)).
+
 ## 6.1.4
 
 - When available return the floor on which the devices is located (see [#562](https://github.com/Baseflow/flutter-geolocator/issues/562));

--- a/geolocator/ios/Classes/Utils/LocationMapper.m
+++ b/geolocator/ios/Classes/Utils/LocationMapper.m
@@ -16,9 +16,11 @@
     double timestamp = [LocationMapper currentTimeInMilliSeconds:location.timestamp];
     double speedAccuracy = 0.0;
     
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
     if (@available(iOS 10.0, *)) {
         speedAccuracy = location.speedAccuracy;
     }
+#endif
     
     NSMutableDictionary *locationMap = [[NSMutableDictionary alloc]initWithCapacity:9];
     [locationMap setObject:@(location.coordinate.latitude) forKey: @"latitude"];

--- a/geolocator/pubspec.yaml
+++ b/geolocator/pubspec.yaml
@@ -1,6 +1,6 @@
 name: geolocator
 description: Geolocation plugin for Flutter. This plugin provides a cross-platform (iOS, Android) API for generic location (GPS etc.) functions.
-version: 6.1.4
+version: 6.1.4+1
 homepage: https://github.com/Baseflow/flutter-geolocator/tree/master/geolocator
 
 environment:

